pantheon-bundle/src/main/java/com/redhat/pantheon/servlet/module/UnpublishVersion.java

```diff
@@ -99,7 +106,20 @@ public void run(SlingHttpServletRequest request, PostResponse response, SlingPos
     @Override
     protected void doRun(SlingHttpServletRequest request, PostResponse response, List<Modification> changes) throws RepositoryException{
         try {
-            ResourceResolver serviceResourceResolver = serviceResourceResolverProvider.getServiceResourceResolver();
+            boolean canUnPublish = false;
+            Session session = request.getResourceResolver().adaptTo(Session.class);
+            UserManager userManager = AccessControlUtil.getUserManager(session);
+            Iterator<Group> groupIterator = userManager.getAuthorizable(session.getUserID()).memberOf();
+            while (groupIterator.hasNext()) {
+                Authorizable group = groupIterator.next();
+                if (group.isGroup() && PantheonConstants.PANTHEON_PUBLISHERS.equalsIgnoreCase(group.getID())) {
+                    canUnPublish = true;
```

Comment:
It will be better to break once you set canPublish as check is done and we have the required state.